src/utilities/utilities.jl

```diff
@@ -11,3 +11,28 @@ function _max(x::Union{Nothing,Any}, y)
         return max(x, y)
     end
 end
+
+function with_temp_dir(f::Function; kwargs...)
+    original_dir = pwd()
+    tmp_dir = mktempdir(; kwargs...)
+    atexit(() -> rm(tmp_dir; force=true, recursive=true))
+
+    cd(tmp_dir)
+    result = f(tmp_dir)
+    cd(original_dir)
+
+    rm(tmp_dir; force=true, recursive=true)
+    return result
+end
+
+function add_compat_section!(project::Dict)
+    if !haskey(project, "compat")
+        project["compat"] = Dict{Any, Any}()
+    end
```

Comment:
```suggestion
    end
    
    return project
```

Reply:
You don't necessarily need to return project since it's a `!` function so it modifies it in place. I could just return nothing, unless the standard for `!` functions are to also return the thing

Reply:
Looked it up `It is typical for such functions to also return the modified array for convenience.` looks like that is the standard.